### PR TITLE
Update OpenAPI docs for .NET 9

### PR DESCRIPTION
**Pull request by captainsafia**
Part of https://github.com/dotnet/AspNetCore.Docs/issues/32477.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/fundamentals/minimal-apis/openapi.md](https://github.com/dotnet/AspNetCore.Docs/blob/753f963d94c4860278a5929e23a562c84547ac22/aspnetcore/fundamentals/minimal-apis/openapi.md) | [OpenAPI support in minimal API apps](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/openapi?branch=pr-en-us-32515) |

### CODE REVIEW

File: aspnetcore/fundamentals/minimal-apis/includes/openapi9.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+When setting the response type for endpoints that may return a ProblemDetails response, the [`ProducesProblem`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.producesproblem) extension method or [`TypedResults.Problem`](/dotnet/api/microsoft.aspnetcore.http.typedresults.problem?) can be used to add the appropriate annotation to the endpoint's metadata.
+
+When there are no explicit annotations provided by one of the strategies above, the framework attempts to determine a default response type by examining the signature of the response. This default response is populated under the `200` status code in the OpenAPI definition.
+
+### Multiple response types
+
+If an endpoint can return different response types in different scenarios, you can provide metadata in the following ways:
+
+* Call the [`Produces`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.produces) extension method multiple times, as shown in the following example:
```
Comment (tdykstra) [inline aspnetcore/fundamentals/minimal-apis/includes/openapi9.md:70]:
```suggestion
* Call the [`Produces`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.produces) extension method multiple times, as shown in the following example:
```

File: aspnetcore/fundamentals/minimal-apis/includes/openapi9.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+  [!code-csharp[](~/fundamentals/minimal-apis/samples/todo/Program.cs?name=snippet_getCustom)]
+
+* Use [`Results<TResult1,TResult2,TResultN>`](xref:Microsoft.AspNetCore.Http.HttpResults.Results%606) in the signature and [`TypedResults`](/dotnet/api/microsoft.aspnetcore.http.typedresults) in the body of the handler, as shown in the following example:
```
Comment (tdykstra) [inline aspnetcore/fundamentals/minimal-apis/includes/openapi9.md:74]:
```suggestion
* Use [`Results<TResult1,TResult2,TResultN>`](xref:Microsoft.AspNetCore.Http.HttpResults.Results%606) in the signature and <xref:Microsoft.AspNetCore.Http.TypedResults> in the body of the handler, as shown in the following example:
```

File: aspnetcore/fundamentals/minimal-apis/includes/openapi9.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  The `Results<TResult1,TResult2,TResultN>` [union types](https://en.wikipedia.org/wiki/Union_type) declare that a route handler returns multiple `IResult`-implementing concrete types, and any of those types that implement `IEndpointMetadataProvider` will contribute to the endpoint’s metadata.
+
+  The union types implement implicit cast operators. These operators enable the compiler to automatically convert the types specified in the generic arguments to an instance of the union type. This capability has the added benefit of providing compile-time checking that a route handler only returns the results that it declares it does. Attempting to return a type that isn't declared as one of the generic arguments to `Results<TResult1,TResult2,TResultN>` results in a compilation error.
+
+## Describe request body and parameters
+
+In addition to describing the types that are returned by an endpoint, OpenAPI also supports annotating the inputs that are consumed by an API. These inputs fall into two categories:
+
+* Parameters that appear in the path, query string, headers, or cookies
+* Data transmitted as part of the request body
```
Comment (tdykstra) [inline aspnetcore/fundamentals/minimal-apis/includes/openapi9.md:87]:
```suggestion
* Parameters that appear in the path, query string, headers, or cookies.
* Data transmitted as part of the request body.
```

File: aspnetcore/fundamentals/minimal-apis/includes/openapi9.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+The framework infers the types for request parameters in the path, query, and header string automatically based on the signature of the route handler.
+
+To define the type of inputs transmitted as the request body, configure the properties by using the [`Accepts`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.accepts) extension method to define the object type and content type that are expected by the request handler. In the following example, the endpoint accepts a `Todo` object in the request body with an expected content-type of `application/xml`.
```
Comment (tdykstra) [inline aspnetcore/fundamentals/minimal-apis/includes/openapi9.md:91]:
```suggestion
To define the type of inputs transmitted as the request body, configure the properties by using the <xref:Microsoft.AspNetCore.Http.OpenApiRouteHandlerBuilderExtensions.Accepts%2A> extension method to define the object type and content type that are expected by the request handler. In the following example, the endpoint accepts a `Todo` object in the request body with an expected content-type of `application/xml`.
```

File: aspnetcore/fundamentals/minimal-apis/includes/openapi9.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+```csharp
+app.MapPost("/todos/{id}", (int id, Todo todo) => ...)
+  .Accepts<Todo>("application/xml");
+```
+
+In addition to the [`Accepts`](/dotnet/api/microsoft.aspnetcore.http.openapiroutehandlerbuilderextensions.accepts) extension method, A parameter type can describe its own annotation by implementing the [`IEndpointParameterMetadataProvider`](/dotnet/api/microsoft.aspnetcore.http.metadata.iendpointparametermetadataprovider) interface. For example, the following `Todo` type adds an annotation that requires a request body with an `application/xml` content-type.
```
Comment (tdykstra) [inline aspnetcore/fundamentals/minimal-apis/includes/openapi9.md:98]:
```suggestion
In addition to the <xref:Microsoft.AspNetCore.Http.OpenApiRouteHandlerBuilderExtensions.Accepts%2A>extension method, a parameter type can describe its own annotation by implementing the <xref:Microsoft.AspNetCore.Http.Metadata.IEndpointParameterMetadataProvider> interface. For example, the following `Todo` type adds an annotation that requires a request body with an `application/xml` content-type.
```

File: aspnetcore/fundamentals/minimal-apis/includes/openapi9.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+        builder.Metadata.Add(new ConsumesAttribute(typeof(Todo), isOptional: false, "application/xml"));
+    }
+}
+```
+
+When no explicit annotation is provided, the framework attempts to determine the default request type if there's a request body parameter in the endpoint handler. The inference uses the following heuristics to produce the annotation:
+
+* Request body parameters that are read from a form via the [`[FromForm]`](/dotnet/api/microsoft.aspnetcore.mvc.fromformattribute) attribute are described with the `multipart/form-data` content-type.
+* All other request body parameters are described with the `application/json` content-type.
+* The request body is treated as optional if it's nullable or if the [`AllowEmpty`](/dotnet/api/microsoft.aspnetcore.mvc.frombodyattribute.microsoft-aspnetcore-http-metadata-ifrombodymetadata-allowempty) property is set on the [`FromBody`](/dotnet/api/microsoft.aspnetcore.mvc.frombodyattribute) attribute.
```
Comment (tdykstra) [inline aspnetcore/fundamentals/minimal-apis/includes/openapi9.md:114]:
```suggestion
* Request body parameters that are read from a form via the [`[FromForm]`](xref:Microsoft.AspNetCore.Mvc.FromFormAttribute) attribute are described with the `multipart/form-data` content-type.
* All other request body parameters are described with the `application/json` content-type.
* The request body is treated as optional if it's nullable or if the <xref:Microsoft.AspNetCore.Http.Metadata.IFromBodyMetadata.AllowEmpty> property is set on the [`FromBody`](xref:Microsoft.AspNetCore.Mvc.FromBodyAttribute) attribute.
```

**Review by tdykstra — APPROVED** [review]
This looks great - I just had some style and link format suggestions.

You would have to change something in the [base article file](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/fundamentals/minimal-apis/openapi.md
) to get it to build with this content. You can replace the >=9.0 content in the file with this content or delete the >=9.0 content in the article file and add an include reference that refers to this file.

File: aspnetcore/fundamentals/minimal-apis/includes/openapi9.md (hunk 1 above)
Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/includes/openapi9.md:70]: Did you mean to recommend something here? It doesn't look like there is actually a diff in the suggestion.

Comment (captainsafia) [reply aspnetcore/fundamentals/minimal-apis/includes/openapi9.md]: Update: I'm gonna assume this was meant to be a recommendation to make it an xref like the other comments.

**Pull request merged**